Tests/BackendIntegrationTests/OtherIntegrationTests.swift

```diff
@@ -24,6 +24,45 @@ class OtherIntegrationTests: BaseBackendIntegrationTests {
         expect(info.isComputedOffline) == false
     }
 
+    func testGetCustomerInfoReturnsNotModified() async throws {
+        // 1. Fetch user once
+        _ = try await Purchases.shared.customerInfo(fetchPolicy: .fetchCurrent)
+
+        let logger = TestLogHandler()
+
+        // 2. Re-fetch user
+        _ = try await Purchases.shared.customerInfo(fetchPolicy: .fetchCurrent)
+
+        let expectedRequest = HTTPRequest(method: .get,
+                                          path: .getCustomerInfo(appUserID: Purchases.shared.appUserID))
+
+        // 3. Verify response was 304
+        logger.verifyMessageWasLogged(
+            Strings.network.api_request_completed(expectedRequest, httpCode: .notModified)
+        )
+    }
+
+    func testGetCustomerInfoAfterLogInReturnsNotModified() async throws {
```

Comment:
This has exposed a bug in signature validation on some 304 responses. I'm still debugging why.

Reply:
Figured it out: #2668.